File: Segundo/PDOO/Tema 1/Civitas/src/main/java/civitas/TestP1.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;


public class TestP1 {
    
    public static void main(String [] args) {
        for (int i = 0; i<100; i++) {
            System.out.println(Dado.getInstance().quienEmpieza(4));
        }
        
        Dado.getInstance().setDebug(true);
        
        for (int i = 0; i<10; i++) {
            System.out.println("Tirada "+i+" "+Dado.getInstance().tirar());
        }
        
        Dado.getInstance().setDebug(false);
        
        for (int i = 0; i<10; i++) {
            System.out.println("Tirada "+i+" "+Dado.getInstance().tirar());
        }
        
        System.out.println(Dado.getInstance().getUltimoResultado());
        
        if (Dado.getInstance().salgoDeLaCarcel()) {
            System.out.println("He salido de la carcel");
        } else {
            System.out.println("No he salido de la carcel");
        }
        
        System.out.println(TipoCasilla.CALLE+" "+TipoSorpresa.IRCASILLA+" "+EstadosJuego.DESPUES_AVANZAR+" "+OperacionesJuego.AVANZAR);
        
        MazoSorpresas mazo = new MazoSorpresas(true);
        Sorpresa s1 = new Sorpresa();
        Sorpresa s2 = new Sorpresa();
        mazo.alMazo(s1);
        mazo.alMazo(s2);
        mazo.siguiente();
        mazo.inhabilitarCartaEspecial(s2);
        mazo.habilitarCartaEspecial(s2);
        
        while(Diario.getInstance().eventosPendientes()) {
            System.out.println(Diario.getInstance().leerEvento());
        }
        
        Tablero t = new Tablero(5);
        Casilla c1 = new Casilla("Hola0", TipoCasilla.CALLE);
        Casilla c2 = new Casilla("Hola1", TipoCasilla.CALLE);
        Casilla c3 = new Casilla("Hola2", TipoCasilla.CALLE);
        Casilla c4 = new Casilla("Hola3", TipoCasilla.CALLE);
        Casilla c5 = new Casilla("Hola4", TipoCasilla.CALLE);
        Casilla c6 = new Casilla("Hola5", TipoCasilla.CALLE);
        
        t.añadeCasilla(c1);
        t.añadeCasilla(c2);
        t.añadeCasilla(c3);
        t.añadeCasilla(c4);
        t.añadeCasilla(c5);
        t.añadeCasilla(c6);
        t.añadeJuez();
        int tirada = Dado.getInstance().tirar();
        System.out.println(tirada);
        System.out.println(t.nuevaPosicion(7, tirada));
    }
    
}

File: Segundo/PDOO/CivitasRuby/Civitas/TestP1.rb
# encoding:utf-8

require_relative "Casilla"
require_relative "Dado"
require_relative "Diario"
require_relative "Estados_juego"
require_relative "MazoSorpresas"
require_relative "Operaciones_juego"
require_relative "Sorpresas"
require_relative "Tablero"
require_relative "TipoCasilla"
require_relative "TipoSorpresa"

module Civitas
    for i in 0..99
        puts Dado.instance.quienEmpieza(4).to_s
    end

    Dado.instance.setDebug(true)
    for i in 0..9
        puts Dado.instance.tirar().to_s
    end
    
    Dado.instance.setDebug(false)
    for i in 0..9
        puts Dado.instance.tirar().to_s
    end

    puts Dado.instance.ultimoResultado.to_s

    if (Dado.instance.salgoDeLaCarcel())
        puts "Salgo de la carcel"
    else
        puts "No salgo de la carcel"
    end

    puts TipoCasilla::CALLE.to_s
    puts Operaciones_juegos::AVANZAR.to_s
    puts TipoSorpresa::IRCARCEL.to_s
    puts Estados_juego::INICIO_TURNO.to_s

    mazo = MazoSorpresas.new()
    s1 = Sorpresas.new("S1")
    s2 = Sorpresas.new("S2")

    mazo.alMazo(s1)
    mazo.alMazo(s2)

    puts mazo.siguiente().nombre
    mazo.inhabilitarCartaEspecial(s1)
    mazo.habilitarCartaEspecial(s1)

    while Diario.instance.eventos_pendientes
        puts Diario.instance.leer_evento
    end

    
    c1 = Casilla.new("c1")
    c2 = Casilla.new("c2")
    c3 = Casilla.new("c3")
    c4 = Casilla.new("c4")
    c5 = Casilla.new("c5")
    tablero = Tablero.new(4)
    tablero.añadeCasilla(c1)
    tablero.añadeCasilla(c2)
    tablero.añadeCasilla(c3)
    tablero.añadeCasilla(c4)
    tablero.añadeCasilla(c5)
    tablero.añadeJuez()
    tirar = Dado.instance.tirar()
    puts tirar.to_s
    puts tablero.nuevaPosicion(6, tirar).to_s
end